drivers/eeprom/eeprom_fm24w256.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2021 Caspar Friedrich <c.s.w.friedrich@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <devicetree.h>
+#include <devicetree/gpio.h>
+#include <drivers/eeprom.h>
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+#include <logging/log.h>
+
+#define DT_DRV_COMPAT cypress_fm24w256
+
+// 256-Kbit ferroelectric random access memory logically organized as 32K × 8 bit
+#define SIZE KB(32)
+
+LOG_MODULE_REGISTER(fm24w256, CONFIG_EEPROM_LOG_LEVEL);
+
+struct config {
+	struct i2c_dt_spec i2c_spec;
+	struct gpio_dt_spec wp_spec;
+	bool read_only;
+};
+
+static int configure_pin(const struct gpio_dt_spec *spec)
+{
+	if (!spec->port) {
+		return -ENODEV;
+	}
+
+	if (!device_is_ready(spec->port)) {
+		return -EACCES;
+	}
+
+	return gpio_pin_configure_dt(spec, GPIO_OUTPUT_INACTIVE);
+}
+
+static int init(const struct device *dev)
+{
+	const struct config *config = dev->config;
+	int ret;
+
+	ret = configure_pin(&config->wp_spec);
+	switch (ret) {
+	case 0:
+		break;
+	case -ENODEV:
+		LOG_DBG("Unable to configure write protect pin: Device not ready (%d)", ret);
+		break;
+	default:
+		LOG_ERR("Unable to configure write protect pin: %d", ret);
+		return ret;
+	}
```

Comment:
I think that it is redundant to verbosely log meaning of error code here, should be enough to log, uniquely across source,  what caused problem and an error code.